tensorflow/core/kernels/data/range_dataset_op_test.cc

```diff
@@ -0,0 +1,207 @@
+        opts, thread_pool, nullptr /* cluster_flr */));
+
+    flr_ = pflr_->GetFLR("/job:localhost/replica:0/task:0/cpu:0");
+
+    if (thread_pool == nullptr) {
+      runner_ = [](std::function<void()> fn) { fn(); };
+    } else {
+      runner_ = [thread_pool](std::function<void()> fn) {
+        thread_pool->Schedule(std::move(fn));
+      };
+    }
+
+    return Status::OK();
+  }
+
+  Status GetDatasetOutputFromContext(int output_index) {
+    auto* tensor = GetOutput(output_index);
+    TF_CHECK_OK(GetDatasetFromVariantTensor(*tensor, &dataset_));
+    return Status::OK();
+  }
+
+  Status RunOpKernel() {
+    // Make sure the old OpKernelContext is deleted before the Params it was
+    // using.
+    context_.reset(nullptr);
+    params_.reset(new OpKernelContext::Params);
+    params_.get()->device = device_.get();
+    params_.get()->frame_iter = FrameAndIter(0, 0);
+    params_.get()->inputs = &inputs_;
+    params_.get()->op_kernel = kernel_.get();
+    params_.get()->function_library = flr_;
+    params_.get()->runner = &runner_;
+
+    step_container_.reset(new ScopedStepContainer(0, [](const string&) {}));
+    params_->step_container = step_container_.get();
+    std::vector<AllocatorAttributes> attrs;
+    test::SetOutputAttrs(params_.get(), &attrs);
+    checkpoint::TensorSliceReaderCacheWrapper slice_reader_cache_wrapper;
+    params_.get()->slice_reader_cache = &slice_reader_cache_wrapper;
+    params_.get()->resource_manager = device_.get()->resource_manager();
+
+    context_.reset(new OpKernelContext(params_.get()));
+    device_->Compute(kernel_.get(), context_.get());
+    return context_->status();
+  }
+
+ protected:
+  template <typename T>
+  void MakeOpDef() {
+    DataType value_type = tensorflow::DataTypeToEnum<T>::value;
+    std::vector<PartialTensorShape>* shapes =
+        new std::vector<PartialTensorShape>({{}});
+    DataTypeVector* dtypes = new DataTypeVector({value_type});
+
+    TF_CHECK_OK(NodeDefBuilder("rangedataset", "RangeDataset")
+                    .Input(FakeInput(DT_INT64))
+                    .Input(FakeInput(DT_INT64))
+                    .Input(FakeInput(DT_INT64))
+                    .Attr("output_types", *dtypes)
+                    .Attr("output_shapes", *shapes)
+                    .Finalize(node_def()));
+    TF_ASSERT_OK(InitOp());
+  }
+
+  Status MakeDataset(int64 start, int64 end, int64 step, int output_index = 0,
+                     int thread_num = 2) {
+    AddInputFromArray<int64>(TensorShape({}), {start});
+    AddInputFromArray<int64>(TensorShape({}), {end});
+    AddInputFromArray<int64>(TensorShape({}), {step});
+
+    TF_CHECK_OK(InitThreadPool(thread_num));
+    TF_CHECK_OK(InitFunctionLibraryRuntime({}, thread_pool_));
+    TF_CHECK_OK(RunOpKernel());
+
+    TF_CHECK_OK(GetDatasetOutputFromContext(output_index));
+    return Status::OK();
+  }
+
+  Status MakeIteratorContext() {
+    iteratorContext_ = new IteratorContext(context_.get());
```

Comment:
You are leaking this memory. `iterator_context_` should be `std::unique_ptr<IteratorContext>`

Reply:
also, I think you could just merge this method into `MakeIterator`